pyincore/utils/dataprocessutil.py

```diff
@@ -123,55 +154,114 @@ def get_mapped_result_from_path(inventory_path: str, dmg_result_path: str,
+            .count()
+            .reset_index()
+        )
+
+        group_by = (
+            mapped_df.groupby(by=[groupby_col_name, "cluster", "category"])
+            .count()
+            .reset_index()
+        )
         group_by = group_by.loc[:, ["guid", groupby_col_name, "cluster", "category"]]
         group_by.rename(columns={"guid": "count"}, inplace=True)
 
-        pivot = group_by.pivot_table(values="count", index=["cluster", "category"], columns=groupby_col_name,
-                                     fill_value=0)
+        pivot = group_by.pivot_table(
+            values="count",
+            index=["cluster", "category"],
+            columns=groupby_col_name,
+            fill_value=0,
+        )
 
         table = pd.DataFrame()
         table[["category", "cluster"]] = unique_categories[["category", "cluster"]]
-        result_by_cluster = pd.merge(table, pivot, how="left", on=["cluster", "category"])
+        result_by_cluster = pd.merge(
+            table, pivot, how="left", on=["cluster", "category"]
+        )
 
         # Add missing max damage states. Handles case when no inventory fall under some damage states.
-        result_by_cluster = result_by_cluster.reindex(result_by_cluster.columns.union(
-            dmg_states, sort=False), axis=1, fill_value=0)
+        result_by_cluster = result_by_cluster.reindex(
+            result_by_cluster.columns.union(dmg_states, sort=False),
+            axis=1,
+            fill_value=0,
+        )
+
+        result_by_category = (
+            result_by_cluster.groupby(by=["category"], sort=False)
+            .sum(min_count=1)
+            .reset_index()
+        )
+
+        result_by_cluster[dmg_states] = (
+            result_by_cluster[dmg_states].fillna(-1).astype(int)
+        )
+        result_by_category[dmg_states] = (
+            result_by_category[dmg_states].fillna(-1).astype(int)
+        )
+
+        return result_by_cluster, result_by_category
 
-        result_by_category = result_by_cluster.groupby(by=["category"], sort=False).sum(min_count=1).reset_index()
+    @staticmethod
+    def create_mapped_dmg_result(
+        inventory,
+        dmg_result,
+        arch_mapping,
+        groupby_col_name="max_state",
+        arch_col="archetype",
+    ):
+        """
 
-        result_by_cluster[dmg_states] = result_by_cluster[dmg_states].fillna(-1).astype(int)
-        result_by_category[dmg_states] = result_by_category[dmg_states].fillna(-1).astype(int)
+        Args:
+            inventory: dataframe represent inventory
+            dmg_result: dmg_result that need to merge with inventory and be grouped
+            arch_mapping: Path to the archetype mappings
+
+        Returns:
+            ret_json: JSON of the results ordered by cluster and category.
+
+        """
+        result_by_cluster, result_by_category = DataProcessUtil.create_mapped_dmg(
+            inventory, dmg_result, arch_mapping, groupby_col_name, arch_col
+        )
```

Comment:
Could we also do the renaming here; so frontend can just directly render the column header as it is
```
result_by_cluster.rename(
            columns={
                "DS_0": "Insignificant",
                "DS_1": "Moderate",
                "DS_2": "Extensive",
                "DS_3": "Complete",
            },
            inplace=True,
        )
        result_by_category.rename(
            columns={
                "DS_0": "Insignificant",
                "DS_1": "Moderate",
                "DS_2": "Extensive",
                "DS_3": "Complete",
            },
            inplace=True,
        )
```

Reply:
I created another frontend task related to this
https://github.com/IN-CORE/incore-playbook/issues/285
If you also want to work on that, feel free to work on that; I put it in the next week's sprint. :-)

Reply:
yep, I can work on it next.